akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1064,6 +1064,29 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
     new Flow(delegate.keepAlive(maxIdle, () ⇒ injectedElem.create()))
 
   /**
+   * Sends elements downstream after each time interval. In other words, this stage set the maximum rate
+   * for emitting messages.
+   *
+   * This stage is `zipWith` combination of upstream connection and [[akka.stream.scaladsl.Source#tick]]
+   * that triggers sending downstream messages.
+   *
+   * General recommendation is to use upstream `buffer` with predefined [[akka.stream.OverflowStrategy]]
+   * that defines behaviour on buffer overflow.
+   *
+   * '''Emits when''' upstream emits an element and configured time elapsed
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @see [[#buffer]], [[akka.stream.scaladsl.Source#tick]].
+   */
+  def throttle(initialDelay: FiniteDuration, interval: FiniteDuration): javadsl.Flow[In, Out, Mat] =
```

Comment:
Also, I think as an advanced use-case, we should have a version that takes a mode:

    throttle(cost = 10, per = 1.second, maximumBurst = 100, costCalculation: (T) => Int, mode: RateLimitMode)

The modes can be:

   - Shaping: this is the backpressured variant. 
   - Enforcing: this variant errors if the limit is reached

The following should work:

     shaping.via(enforcing) should not fail (i.e. if user shapes its traffic according to SLA then server will not kick him out)

There are fun uses of this btw:

    // 256kBytes/sec on average, with a 256kByte allowed initial burst
    tcpBytes.throttle(cost = 256 * 1024, 1.second, burst = 256 * 1024(byteString) => byteString.size)
      .via(parser)
      .throttle(cost = 100, 1.minute, burst = 10) // and then allowing only 100 API calls/minute (with some burstyness allowed)

Which can work for large requests where we want to limit both the datarate and the count of requests.